geode-core/src/main/java/org/apache/geode/cache/Region.java

```diff
@@ -1589,6 +1601,93 @@ Object selectValue(String queryPredicate) throws FunctionDomainException, TypeMi
    */
   void registerInterest(K key, InterestResultPolicy policy);
 
+
+
+  default void registerInterestForAllKeys() {
+    registerInterestRegex(".*");
+  }
+
+
+  /**
+   * Sends a request to the CacheServer to register interest in for all keys for this client.
+   * Updates to this key by other clients will be pushed to this client by the CacheServer. This
+   * method is currently supported only on clients in a client server topology. This key is first
+   * locally cleared from the client and current value for this key is inserted into the local cache
+   * before this call returns. (if requested).
+   *
+   * <p>
+   * If you locally-destroy a key and your region has concurrency-checks-enabled turned off you will
+   * not receive invalidation events from your interest subscription for that key. When
+   * concurrency-checks-enabled is turned on GemFire will accept invalidation and deliver these
+   * events to your client cache.
+   * </p>
+   *
+   * @param key The key on which to register interest. The following <code>List</code> and
```

Comment:
This javadoc has a couple warnings. key isn't an argument, and policy is called interestPolicy